### PR TITLE
Additional parameters added for ssh_config

### PR DESCRIPTION
**Pull request by apazga**
Hi!

I just added the following parameters (README.md updated & parameter validation added too).
* ssh_strict_host_key_checking
* ssh_enable_ssh_keysign
* ssh_hostbasedauthentication

Best regards!
Abel.

### CODE REVIEW

File: spec/fixtures/ssh_config_debian [hunk 1]
```diff
@@ -59,3 +59,4 @@ Host *
   SendEnv LANG LANGUAGE LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES
   SendEnv LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT
   SendEnv LC_IDENTIFICATION LC_ALL
+#  EnableSSHKeysign no
```
Comment (ghoneycutt) [inline spec/fixtures/ssh_config_debian]: The fixture should not change as they are defaults for the OS

File: templates/ssh_config.erb [hunk 2]
```diff
@@ -90,3 +94,7 @@ GSSAPIDelegateCredentials <%= @ssh_gssapidelegatecredentials %>
 <% if @ssh_config_macs -%>
   MACs <%= @ssh_config_macs.join(',') %>
 <% end -%>
+#  EnableSSHKeysign no
```
Comment (ghoneycutt) [inline templates/ssh_config.erb]: Move this into the conditional statement below

File: manifests/init.pp [hunk 3]
```diff
@@ -488,6 +491,14 @@
     validate_re($sshd_gssapicleanupcredentials_real, '^(yes|no)$', "ssh::sshd_gssapicleanupcredentials may be either 'yes' or 'no' and is set to <${sshd_gssapicleanupcredentials_real}>.")
   }
 
+  if $ssh_strict_host_key_checking != undef {
+    validate_re($ssh_strict_host_key_checking, '^(yes|no)$', "ssh::ssh_ssh_strict_host_key_checking may be either 'yes' or 'no' and is set to <${ssh_strict_host_key_checking}>.")
```
Comment (ghoneycutt) [inline manifests/init.pp]: Valid values are yes, no and ask.

**Comment by ghoneycutt** [comment]
This module also requires spec tests. There are other examples of parameters that are very similar, so hopefully most of the work is copy/paste and change some param names.

**Comment by apazga** [comment]
Done! (I think). If I missed something just tell me!
Hope it helps!!

**Comment by apazga** [comment]
Ooops, I'll check it out again later, it's seems some tests are not ok.

**Comment by ghoneycutt** [comment]
Hi @apazga 

Funny profile pic :)

I worked on your PR and made some changes to get it all working in PR #168

**Comment by apazga** [comment]
^_^ Thanks @ghoneycutt! You rocks!

**Pull request closed without merging**